lib/ansible/parsing/dataloader.py

```diff
@@ -326,12 +325,12 @@ def path_dwim_relative(self, path: str, dirname: str, source: str, is_role: bool
             search.append(self.path_dwim(source))
 
         for candidate in search:
-            if os.path.exists(to_bytes(candidate, errors='surrogate_or_strict')):
+            if os.path.exists(candidate):
                 break
 
         return candidate
 
-    def path_dwim_relative_stack(self, paths: list[str], dirname: str, source: str, is_role: bool = False) -> str:
+    def path_dwim_relative_stack(self, paths: list[str], dirname: str | None, source: str, is_role: bool = False) -> str:
```

Comment:
Seems `source` can be `None` as well. Checks for it below.

```suggestion
    def path_dwim_relative_stack(self, paths: list[str], dirname: str | None, source: str | None, is_role: bool = False) -> str:
```